tests/v2/validation/charts/cisbenchmark_test.go

```diff
@@ -0,0 +1,298 @@
+	c.project = createdProject
+
+	c.chartInstallOptions = &charts.InstallOptions{
+		ClusterName: clusterName,
+		ClusterID:   clusterID,
+		Version:     cisConfig.ChartVersion,
+		ProjectID:   createdProject.ID,
+	}
+
+}
+
+func (c *CisBenchmarkTestSuite) TestCisBenchmarkChart() {
+	subSession := c.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := c.client.WithSession(subSession)
+	require.NoError(c.T(), err)
+
+	c.T().Log("Installing latest version of cis benchmark chart")
+	err = charts.InstallRancherCisBenchmarkChart(client, c.chartInstallOptions)
+	require.NoError(c.T(), err)
+
+	c.T().Log("Waiting for cisbenchmark chart deployments to have expected number of available replicas")
+	err = charts.WatchAndWaitDeployments(client, c.project.ClusterID, charts.RancherCisBenchmarkNamespace, metav1.ListOptions{})
+	require.NoError(c.T(), err)
+
+	scanName := namegen.AppendRandomString("scan")
+
+	c.T().Log("Running a scan")
+	yamlInput := &management.ImportClusterYamlInput{
+		DefaultNamespace: charts.RancherCisBenchmarkNamespace,
+		YAML:             fmt.Sprintf(ClusterScanYaml, scanName, c.cisProfileName),
+	}
+
+	cluster, err := client.Management.Cluster.ByID(c.project.ClusterID)
+	require.NoError(c.T(), err)
+	_, err = client.Management.Cluster.ActionImportYaml(cluster, yamlInput)
+	require.NoError(c.T(), err)
+
+	steveClient, err := client.Steve.ProxyDownstream(c.project.ClusterID)
+	require.NoError(c.T(), err)
+
+	err = wait.Poll(1*time.Second, 5*time.Minute, func() (done bool, err error) {
+		scanList, err := steveClient.SteveType(charts.ClusterScanResourceType).List(nil)
+		if err != nil {
+			return false, err
+		}
+
+		scanIdx := -1
+
+		for idx, scan := range scanList.Data {
+			if scan.Name == scanName {
+				scanIdx = idx
+				break
+			}
+		}
+
+		if scanIdx < 0 {
+			return false, errors.New("scan not found")
+		}
+
+		clusterScanStatusType := &charts.ClusterScanStatus{}
+		scanStatus := scanList.Data[scanIdx].Status
+		err = v1.ConvertToK8sType(scanStatus, clusterScanStatusType)
+		if err != nil {
+			return false, err
+		}
+
+		if clusterScanStatusType.Display.State == scanError {
+
+			return false, errors.New(clusterScanStatusType.Display.Message)
+		} else if clusterScanStatusType.Display.State == scanPending ||
+			clusterScanStatusType.Display.State == scanRunning ||
+			clusterScanStatusType.Display.State == scanReporting {
+
+			logrus.Infof("cluster scan is %s", clusterScanStatusType.Display.State)
+			return false, nil
+		}
+
+		return true, nil
+	})
+	require.NoError(c.T(), err)
+
+	c.CheckScanReport(steveClient, scanName)
+}
+
+func (c *CisBenchmarkTestSuite) CheckScanReport(steveClient *v1.Client, scanName string) {
+	scanReportList, err := steveClient.SteveType(charts.ClusterScanReportType).List(nil)
+	require.NoError(c.T(), err)
+
+	scanReportIdx := -1
+
+	for idx, scanReport := range scanReportList.Data {
+		if strings.Contains(scanReport.Name, scanName) {
+			scanReportIdx = idx
+			break
+		}
+	}
```

Comment:
this concept is used more than once, it might make sense to pull it out into its own helper.

Reply:
Thanks! I updated it.